### ISSUE TITLE
App does not accept Bitcoin-Core connection

### ISSUE THREAD

**Issue by SebiCreator** [sparrowwallet/sparrow#737]
In my settings i configured the Server as a bitcoin-core server on a remote machine. If i test the connection in settings it all works as it should. 
<img width="743" alt="Bildschirm­foto 2022-11-05 um 17 47 24" src="https://user-images.githubusercontent.com/63467522/200132038-7b8fa88d-f079-4c55-8c41-151c1e3941dd.png">

But outside of the settings the app always shows "Connection failed: Server Error (Check if Bitcoin Core is running and the authentication details are correct".
<img width="722" alt="Bildschirm­foto 2022-11-05 um 17 47 47" src="https://user-images.githubusercontent.com/63467522/200132040-22ada587-c836-462b-b3f0-9324091c61de.png">

The logging from Bitcoin-Core shows the following 
2022-11-05T16:50:50Z [sparrow] External scriptPubKey Manager for output type 0 does not exist
2022-11-05T16:50:50Z [sparrow] External scriptPubKey Manager for output type 1 does not exist
2022-11-05T16:50:50Z [sparrow] External scriptPubKey Manager for output type 2 does not exist
2022-11-05T16:50:50Z [sparrow] Internal scriptPubKey Manager for output type 0 does not exist
2022-11-05T16:50:50Z [sparrow] Internal scriptPubKey Manager for output type 1 does not exist
2022-11-05T16:50:50Z [sparrow] Internal scriptPubKey Manager for output type 2 does not exist
...
<img width="658" alt="Bildschirm­foto 2022-11-05 um 17 51 01" src="https://user-images.githubusercontent.com/63467522/200132052-0f0bac7f-1d92-495c-9fd5-5bdf21773ed8.png">

I dont know how to interpret this information.
To me it looks like a bug?

Does somebody know what's the problem and how to solve it?

**Comment by gStart9** [comment]
![sparrow-electrs-on_bitcoind_rpc-why](https://user-images.githubusercontent.com/106188942/200141640-c97a6d93-66a6-49cc-9afb-aef6fe860acd.jpg)
I am experiencing this too.  Here, I put in my .onion of my bitcoin RPC, and yet it (suppsedly) connects to an electrs somewhere -- but where?  I'm not running electrs!

**Comment by craigraw** [comment]
@SebiCreator this appears to be an issue related to Bitcoin Core, not Sparrow. Do you have any unusual configuration of Bitcoin Core? See this link too: https://bitcoin.stackexchange.com/questions/96415/wallet-files-are-empty

@gStart9 You are not starting Electrs. Sparrow contains a library called BWT which creates an Electrum server backed by a Bitcoin Core wallet. Sparrow then communicates with the network via that Electrum server. Electrs is another implementation of an Electrum server. Note that communicating with Bitcoin Core using BWT over Tor is not recommended with Tor being as slow as it is currently - the architecture does not lend itself to this, and there are privacy issues besides (anyone with your onion address can see your balance and control your node).

**Comment by SebiCreator** [comment]
@craigraw thanks for your answer.
This is my bitcoin.conf file.
<img width="542" alt="Bildschirm­foto 2022-11-06 um 08 56 47" src="https://user-images.githubusercontent.com/63467522/200160344-a5658854-9d56-4915-b62a-1bee856f3883.png">

I start the daemon with `sudo bitcoind -conf=/mnt/ssd/mainnet/bitcoin.conf`

Any problems with this?

**Comment by craigraw** [comment]
It looks ok to me. Try shutting down Bitcoin Core and Sparrow, and deleting the 'sparrow' wallet in the Bitcoin Core wallets folder. Sparrow will recreate it. It may also help to upgrade Bitcoin Core, but that's less likely. If you do, upgrade Sparrow first if you haven't already.

**Comment by gStart9** [comment]
> @gStart9 You are not starting Electrs. Sparrow contains a library called BWT which creates an Electrum server backed by a Bitcoin Core wallet. Sparrow then communicates with the network via that Electrum server. Electrs is another implementation of an Electrum server. Note that communicating with Bitcoin Core using BWT over Tor is not recommended with Tor being as slow as it is currently - the architecture does not lend itself to this, and there are privacy issues besides (anyone with your onion address can see your balance and control your node).

So these `Electrum Server ready` things are coming from the bwt library running within Sparrow?  How can I tell if I'm getting through to the .onion bitcoin rpc?  The test never seems to fully indicate.

So in other words, it sounds like sparrow doesn't speak to the bitcoin rpc directly, but speaks electrum server protocol to bwt which translates the calls to the bitcoin rpc?  Understood about tor sucking right now, but in theory, .onion can work, correct?

**Comment by craigraw** [comment]
> So in other words, it sounds like sparrow doesn't speak to the bitcoin rpc directly, but speaks electrum server protocol to bwt which translates the calls to the bitcoin rpc? Understood about tor sucking right now, but in theory, .onion can work, correct?

Correct - I suspect some of the required calls are not completing over Tor. Unfortunately with the current Tor issues this approach is going to be a poor user experience. Connecting to an Electrum server over Tor requires far fewer calls and is still a reasonable approach, even with the current DDos attack.

**Comment by turkycat** [comment]
+1 this issue while trying to set up Sparrow for the first time today. I'm having the exact same experience as OP, but I'm not using TOR. I'm also not running Electrs.

**Comment by SebiCreator** [comment]
> It looks ok to me. Try shutting down Bitcoin Core and Sparrow, and deleting the 'sparrow' wallet in the Bitcoin Core wallets folder. Sparrow will recreate it. It may also help to upgrade Bitcoin Core, but that's less likely. If you do, upgrade Sparrow first if you haven't already.

Deleted sparrow + metadata, deleted sparrow Wallet on Bitcoin Core. First setup with bitcoin core works, but at the moment i imported my first wallet, the old problems appears.

Are there anybody who run Sparrow with Bitcoin Core without Problems?

**Comment by craigraw** [comment]
Yes, Taproot wallets are not supported when connecting to Bitcoin Core. See https://github.com/bwt-dev/bwt/issues/91 to view the upstream issue on this. It's frustrating that some projects become blocking dependencies on certain functionality, but that's unfortunately the case here.

I was hoping this problem could be resolved more quickly, but since it doesn't look like rust-bitcoincore-rpc is getting any regular updates, I've added an error message to warn of the incompatibility in a25b53bd (and [a comment on the relevant issue](https://github.com/rust-bitcoin/rust-bitcoincore-rpc/pull/199#issuecomment-1306817195)!)

Closing this off as I believe the original issue has been identified.
